docs/design/Ray-Job-Manager.md

```diff
@@ -0,0 +1,78 @@
+# Ray TPU Job Manager
+
+This is a quick design document to explain how our Ray TPU Job Manager works.
+
+## Introduction
+
+Please see the [Ray documentation](https://docs.ray.io/en/latest/index.html) for more information.
+Ray is a resource-aware job scheduler, so you can specify the resources that a job requires:
+
+```python
+@ray.remote(num_cpus=4)
+def my_cpu_job():
+    ...
+```
+
+For GPUs, Ray lets you specify the number of GPUs you need:
+
+```python
+@ray.remote(num_gpus=1)
+def my_gpu_job():
+    ...
+```
+
+In Ray, TPUs are roughly represented the same way, but there are a number of problems with that approach.
+In particular:
+
+* Ray's granularity allows it to schedule a task on a single machine, not across multiple machines. In particular,
```

Comment:
does task = job?  if so, choose one?

Reply:
Eh, I use job to mean a training run (that might get preempted) and task to mean a ray.remote invocation

Reply:
i'll clarify